dev/reference/alert-rules.md

```diff
@@ -0,0 +1,1276 @@
+
+    etcd writes data to disk with a slower speed than normal. It might lead to PD leader timeout or failure to store TSO on disk in time, which will shut down the service of the entire cluster.
+
+* Solution:
+
+    * Find the cause of slow writes. It might be other services that overload the system. You can check whether PD itself occupies a large amount of CPU or IO resources.
+    * Try to restart PD or manually transfer leader to another PD to recover the service.
+    * Make offline and replace the PD with issues if it cannot be recovered due to environmental factors.
+
+#### `PD_miss_peer_region_count`
+
+* Alert rule:
+
+    `sum(pd_regions_status{type="miss_peer_region_count"}) > 100`
+
+* Description:
+
+    The number of Region replicas is smaller than the value of `max-replicas`. When the TiKV is down and its downtime exceeds `max-down-time`, it usually leads to missing replicas for some Regions during a period of time. When a TiKV node is made offline, it might result in a small number of Regions with missing replicas.
+
+* Solution:
+
+    * Find the cause of the issue by checking whether there is any TiKV that is down or being made offline.
+    * Watch the region health panel and see whether `miss_peer_region_count` is continuously decreasing.
+
+### Warning-level alerts
+
+Warning-level alerts are a reminder for an issue or error.
+
+#### `PD_cluster_lost_connect_tikv_nums`
+
+* Alert rule:
+
+    `sum(pd_cluster_status{type="store_disconnected_count"}) > 0`
+
+* Description:
+
+    PD does not receive a TiKV heartbeat within 20 seconds. Normally a TiKV heartbeat comes in every 10 seconds.
+
+* Solution:
+
+    * Check whether the TiKV is being restarted.
+    * Check whether the TiKV process is normal, the network is isolated, and the load is too high, and recover the service as much as possible.
+    * If you confirm that the TiKV cannot be recovered, you can make it offline.
+    * If you confirm that the TiKV can be recovered, but not in the short term, you can consider increasing the value of `max-down-time`. It will prevent the TiKV from being considered as irrecoverable and the data from being removed from the TiKV.
+
+#### `PD_cluster_low_space`
+
+* Alert rule:
+
+    `sum(pd_cluster_status{type="store_low_space_count"}) > 0`
+
+* Description:
+
+    Indicates that there is no sufficient space on the TiKV node.
+
+* Solution:
+
+    * Check whether there is no sufficient space in the cluster in general. If so, you need to increase its capacity.
+    * Check whether there is any issue with Region balance scheduling, which leads to unevenly distributed data.
+    * Check whether there is any file that occupies a large amount of disk space, such as the log, snapshot, core dump, etc.
+    * Lower the Region weight of the node to reduce the data volume.
+    * When it is not possible to release the space, you can consider proactively making the node offline. This prevents insufficient disk space that leads to downtime.
```

Comment:
```suggestion
    * When it is not possible to release the space, consider proactively making the node offline to avoid insufficient disk space that leads to downtime.
```